### PR TITLE
Imp 382 falling snow cli

### PR DESCRIPTION
**Pull request by wxtim**
Reference issue if exists

Testing:
 - [ ] Ran tests and they passed OK
 - [ ] Added new tests for the new feature(s)

### CODE REVIEW

File: bin/improver-continuous-snow-falling-level [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to apply thresholding to a cube."""
```
Comment (bayliffe) [inline bin/improver-continuous-snow-falling-level]: Needs updating.

File: bin/improver-continuous-snow-falling-level [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                        "falling snow level is being calculated.")
+    parser.add_argument("pressure", metavar="PRESSURE",
+                        help="File path to a cube of air pressures at"
+                        " heights (m) at the points for which the continuous "
+                        "falling snow level is being calculated.")
+    parser.add_argument("orography", metavar="OROGRAPHY",
+                        help="A path to an input NetCDF file containing "
+                        "a cube with the orography in m of the terrain "
+                        "that the continuous falling snow level is being"
+                        "calculated.")
```
Comment (bayliffe) [inline bin/improver-continuous-snow-falling-level]: `A path to an input NetCDF file containing a cube with the orography in m of the terrain over which the continuous falling snow level is being calculated.`

File: bin/improver-continuous-snow-falling-level [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    parser.add_argument("--precision", metavar="NEWTON_PRECISION",
+                        default=0.005, type=float,
+                        help="Precision to which the wet bulb temperature "
+                        "is required: This is used by the Newton iteration "
+                        "default value is 0.005")
+    parser.add_argument("--falling_level_threshold",
+                        metavar="FALLING_LEVEL_THRESHOLD",
+                        default=90.0, type=float,
+                        help="Cutoff threshold for the wet-bulb integral used "
+                        "to calculate the falling snow level.  "
```
Comment (bayliffe) [inline bin/improver-continuous-snow-falling-level]: `Cutoff threshold for the wet-bulb integral used to calculate the falling snow level. This threshold indicates the level at which falling snow is deemed to have melted to become rain. The default value is 90.0, an empirically derived value.`

File: bin/improver-continuous-snow-falling-level [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                        "Default Value is 90.0")
+
+    args = parser.parse_args()
+
+    precision = args.precision
+    falling_level_threshold = args.falling_level_threshold
```
Comment (bayliffe) [inline bin/improver-continuous-snow-falling-level]: Why not just use the args.<> values as arguments below and remove these additional statements?

File: bin/improver-continuous-snow-falling-level [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    result = FallingSnowLevel(
+        precision=precision,
+        falling_level_threshold=falling_level_threshold).process(
+            temperature,
+            relative_humidity,
+            pressure,
+            orog)
+
+    iris.save(result, args.output_filepath)
```
Comment (bayliffe) [inline bin/improver-continuous-snow-falling-level]: `, unlimited_dimensions=[])`

File: tests/improver-continuous-snow-falling-level/00-null.bats [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+@test "continuous-snow-falling-level no arguments" {
+  run improver continuous-snow-falling-level
+  [[ "$status" -eq 2 ]]
+  expected="usage: improver-continuous-snow-falling-level [-h]
+                                              [--precision NEWTON_PRECISION]
+                                              [--falling_level_threshold FALLING_LEVEL_THRESHOLD]
+                                              TEMPERATURE RELATIVE_HUMIDITY
+                                              PRESSURE OROGRAPHY OUTPUT_FILE
+improver-continuous-snow-falling-level: error: too few arguments"
```
Comment (bayliffe) [inline tests/improver-continuous-snow-falling-level/00-null.bats]: The `too few arguments` line is not required for this comparison to succeed.

File: tests/improver-continuous-snow-falling-level/01-help.bats [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  -h, --help            show this help message and exit
+  --precision NEWTON_PRECISION
+                        Precision to which the wet bulb temperature is
+                        required: This is used by the Newton iteration default
+                        value is 0.005
+  --falling_level_threshold FALLING_LEVEL_THRESHOLD
+                        Cutoff threshold for the wet-bulb integral used to
+                        calculate the falling snow level. Default Value is
+                        90.0
+
```
Comment (bayliffe) [inline tests/improver-continuous-snow-falling-level/01-help.bats]: Unneeded extra line

File: bin/improver-continuous-snow-falling-level (hunk 1 above)
Comment (wxtim) [reply bin/improver-continuous-snow-falling-level]: Done

File: bin/improver-continuous-snow-falling-level (hunk 3 above)
Comment (wxtim) [reply bin/improver-continuous-snow-falling-level]: Done

File: bin/improver-continuous-snow-falling-level (hunk 2 above)
Comment (wxtim) [reply bin/improver-continuous-snow-falling-level]: Done

File: bin/improver-continuous-snow-falling-level (hunk 4 above)
Comment (wxtim) [reply bin/improver-continuous-snow-falling-level]: Done

File: bin/improver-continuous-snow-falling-level (hunk 5 above)
Comment (wxtim) [reply bin/improver-continuous-snow-falling-level]: Done

File: tests/improver-continuous-snow-falling-level/00-null.bats (hunk 6 above)
Comment (wxtim) [reply tests/improver-continuous-snow-falling-level/00-null.bats]: Done

File: tests/improver-continuous-snow-falling-level/01-help.bats (hunk 7 above)
Comment (wxtim) [reply tests/improver-continuous-snow-falling-level/01-help.bats]: Done?

**Comment by benfitzpatrick** [comment]
Maybe a shorter name - e.g. `improver snow-falling-level`?

**Comment by wxtim** [comment]
I'm inclined to agree with what @benfitzpatrick said about the name: It's a bit clunky and I find continuous to be a word whose spelling I have to devote brain power to. :( I'm reluctant to change it prior to having an "in person discussion", however.

**Pull request merged**